app/models/user_access.rb

```diff
@@ -59,6 +59,63 @@ def can?(action, model, record = nil)
     end
   end
 
+  def access_tree(action)
+    facilities = accessible_facilities(action).includes(facility_group: :organization)
+
+    facility_tree =
+      facilities
+        .map { |facility| [facility, {can_access: true}] }
+        .to_h
+
+    facility_group_tree =
+      facilities
+        .map(&:facility_group)
+        .map { |fg|
+          facilities_in_facility_group =
+            facility_tree.select { |facility, _| facility.facility_group == fg }
+
+          [fg,
+            {
+              can_access: can?(action, :facility_group, fg),
+              facilities: facilities_in_facility_group,
+              total_facilities: fg.facilities.size
```

Comment:
We only expose totals, not the actual objects. Being overly defensive here since we're dealing with authorization.